test/testdrive/dependencies.td

```diff
@@ -332,6 +332,80 @@ unknown catalog item 'j2'
 
 > DROP SOURCE s5;
 
+# https://github.com/MaterializeInc/materialize/issues/5577
+> CREATE TYPE int4_list AS LIST (element_type=int4)
+
+> CREATE VIEW v1 AS SELECT CAST('{2}' AS int4_list)
+
+! DROP TYPE int4_list
+cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.v1'
+
+> DROP VIEW v1
+
+> CREATE TABLE t1 (custom int4_list)
+
+! DROP TYPE int4_list
+cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.t1'
+
+> DROP TABLE t1
+
+> SHOW TABLES
+
+> CREATE MATERIALIZED VIEW v1 AS SELECT * FROM ( SELECT CAST('{2}' AS int4_list) )
+
+! DROP TYPE int4_list
+cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.v1'
+
+> DROP VIEW v1
+
+> CREATE VIEW v1 AS SELECT CAST(CAST('{2}' AS int4_list) AS text)
+
+! DROP TYPE int4_list
+cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.v1'
+
+> DROP VIEW v1
+
+> CREATE VIEW v1 AS VALUES (CAST('{2}' AS int4_list))
+
+! DROP TYPE int4_list
+cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.v1'
+
+> DROP VIEW v1
+
+> CREATE VIEW v1 AS SELECT MIN(CAST(CAST('{1}' AS int4_list) AS string))
+
+! DROP TYPE int4_list
+cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.v1'
+
+> DROP VIEW v1
+
+> CREATE TABLE t1 (f1 TEXT DEFAULT CAST ('{}' AS int4_list))
+
+> DROP TABLE t1
+
```

Comment:
```! DROP TYPE``` is missing here, apologies for that. It should throw an error.

Reply:
Oops, thank you!